### PR TITLE
Update docs to better clarify Kafka usage

### LINKED ISSUE #306: Unclear getting-started instruction for Kafka
As a user, while attempting to orchestrate KStreams and Kafka binder based Apps, I couldn't find any clear instructions on how to switch to Kafka. In particular, there is no mention of KStreams binder configs. 

**Acceptance:**
- Though we have the mention of `If you’d like to orchestrate stream processing pipelines with Apache Kafka as the messaging middleware using Skipper, you must change the values in the SPRING_APPLICATION_JSON environment variable:` as a NOTE, it is nor explicit enough for the users. Perhaps it'd be better to split the configs for Kafka and Rabbit each, as separate sections.
- Include a copyable JSON for Rabbit, Kafka, and Kafka + KStreams excplicitly.
- Clarify the reasoning for when Kafka and KStreams are used together.

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #306

### CODE REVIEW

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -65,19 +65,35 @@ $ git clone https://github.com/{github-repo}
 $ cd spring-cloud-dataflow-server-kubernetes
 $ git checkout {github-tag}
 ----
+
+==== Choosing a Transport
+
+For deployed applications to communicate with each other, a transport middleware needs to be selected. The sample deployment and service YAML files provide configurations for RabbitMQ and Kafka. Only one transport needs to be configured.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: "a transport middleware needs to be selected" -> "a message broker needs to be selected"

Comment (sabbyanandan) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Possibly similar change to: "Only one transport needs to be configured"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 2]
```diff
@@ -141,19 +177,21 @@ so it is important to define the "accounts" based on the project preferences. In
...
+----
++
+Additionally, if you would like to use the
+link:https://docs.spring.io/spring-cloud-stream/docs/current/reference/htmlsingle/#_apache_kafka_streams_binder[Apache Kafka Streams Binder], the `SPRING_APPLICATION_JSON` environment variable in `src/kubernetes/skipper/skipper-deployment.yaml` would be configured as follows:
++
+[source,yaml]
+----
+"{\"spring.cloud.skipper.server.enableLocalPlatform\" : false, \"spring.cloud.skipper.server.platform.kubernetes.accounts.minikube.environmentVariables\": \"SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS=${KAFKA_SERVICE_HOST}:${KAFKA_SERVICE_PORT}, SPRING_CLOUD_STREAM_KAFKA_BINDER_ZK_NODES=${KAFKA_ZK_SERVICE_HOST}:${KAFKA_ZK_SERVICE_PORT}, SPRING_CLOUD_STREAM_KAFKA_STREAMS_BINDER_BROKERS=${KAFKA_SERVICE_HOST}:${KAFKA_SERVICE_PORT}, SPRING_CLOUD_STREAM_KAFKA_STREAMS_BINDER_ZK_NODES=${KAFKA_ZK_SERVICE_HOST}:${KAFKA_ZK_SERVICE_PORT}\",\"spring.cloud.skipper.server.platform.kubernetes.accounts.minikube.memory\" : \"1024Mi\",\"spring.cloud.skipper.server.platform.kubernetes.accounts.minikube.createDeployment\" : true}"
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Maybe supply `\"spring.cloud.skipper.server.platform.kubernetes.accounts.minikube.memory\" : \"1024Mi\",\"spring.cloud.skipper.server.platform.kubernetes.accounts.minikube.createDeployment\" : true` here also. Not just the boolean.

Comment (sabbyanandan) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: I think all we need is one property with the prefix `spring.cloud.skipper.server.platform.kubernetes.accounts.minikube`, so we could remove boolean from both the JSON.

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 3]
```diff
@@ -196,19 +234,38 @@ To use Skipper, you _must_ uncomment the following properties to `src/kubernetes
...
 +
+Create Role Bindings and Service account:
++
 ```
 $ kubectl create -f src/kubernetes/server/server-roles.yaml
 $ kubectl create -f src/kubernetes/server/server-rolebinding.yaml
 $ kubectl create -f src/kubernetes/server/service-account.yaml
+```
++
+When using RabbitMQ for transport:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: we can drop "for transport"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 4]
```diff
@@ -471,19 +528,37 @@ Successfully targeted http://130.211.203.246/
 dataflow:>
 ```
 +
-. Register the Docker with Rabbit binder versions of the `time` and `log` apps using the shell.
+. Register the Docker binder versions of the `time` and `log` apps using the shell.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: "Register the Docker binder versions" -> "Register the Docker images of the Rabbit binder based `time` and `log` apps using the shell"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 5]
```diff
@@ -471,19 +528,37 @@ Successfully targeted http://130.211.203.246/
...
 +
 [subs=attributes]
 ```
 dataflow:>app register --type source --name time --uri docker://springcloudstream/time-source-rabbit:{docker-time-source-rabbit-version} --metadata-uri maven://org.springframework.cloud.stream.app:time-source-rabbit:jar:metadata:{docker-time-source-rabbit-version}
 dataflow:>app register --type sink --name log --uri docker://springcloudstream/log-sink-rabbit:{docker-log-sink-rabbit-version} --metadata-uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:jar:metadata:{docker-log-sink-rabbit-version}
 ```
 +
-. Alternatively, if you would like to register all out-of-the-box stream applications built with the Rabbit binder in bulk,
-you can with the following command. For more details, review how to link:http://docs.spring.io/spring-cloud-dataflow/docs/{scdf-core-version}/reference/html/spring-cloud-dataflow-register-apps.html[register applications].
+When using Kafka as the transport:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Drop "as the transport"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 6]
```diff
@@ -537,13 +612,22 @@ Refer to the section <<streams-using-skipper>> for more information.
...
+. Register the `http-source`, you can use one of the following commands:
++
+When using RabbitMQ as the transport:
 +
 [subs=attributes]
 ```
 dataflow:>app register --type source --name http --uri docker//springcloudstream/http-source-rabbit:{docker-http-source-rabbit-version} --metadata-uri maven://org.springframework.cloud.stream.app:http-source-rabbit:jar:metadata:{docker-http-source-rabbit-version}
 ```
 +
+When using Kafka as the transport:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: It'd be good to remove all occurrences of "transport"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc (hunk 2 above)
Comment (chrisjs) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: those values are there. this is uses the same yaml as the exiting (rabbit) config, just kafka vals in place. do you want to change the existing `skipper-deployment.yaml` SAJ block + here or something else?

**Comment by chrisjs** [comment]
updates pushed

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc (hunk 2 above)
Comment (sabbyanandan) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Just one property is enough, so yes, we could change it in the YAML's and the docs.

Comment (chrisjs) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: ok removed `createDeployment` in these docs and the `skipper-deployment.yaml` as it defaults to true in the deployer anyway

**Review by sabbyanandan — APPROVED** [review]
Looks great! Thanks for the revisions.

**Comment by chrisjs** [comment]
merged in 071360a

**Pull request closed without merging**